Mermaid example: onboarding sequence diagram

Silvanoc opened the pull request:
### Description

Replace a diagram added as an image with a diagram generated on-the-fly with Mermaid.

#### Issues Addressed

Diagrams as images are not editable and changes cannot be easily identified.

#### Change Type

Please select the relevant options:

- [ ] Fix (change that resolves an issue)
- [ ] New enhancement (change that adds specification content)
- [x] Content edits (change that edits existing content)

#### Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [x] My changes adhere to the established patterns, and best practices.

Silvanoc committed:
feat(onboarding): small improvements and fixes

Improvements and fixes:

- Uniformize actor activation code in Mermaid sequence diagram.
- Fix erroneous arrows for return messages in onboarding sequence diagram.
- Small readibility improvement in onboarding sequence diagram: Request messages have a continuous line and returns a dashed line.
- Add to onboarding sequence diagram a box showing the proposal for using a FIDO onboarding workflow.
- Improve readability
- Add legend
- Document possible sequence diagram change
- Typos

Signed-off-by: Silvano Cirujano Cuesta <silvano.cirujano-cuesta@siemens.com>

Silvanoc commented:
This is as of now a PR meant to be used as discussion basis, therefore *Draft*. Some parts of it will be discussed directly in this PR. Micro-commits for easy understanding of the changes should be squashed before merging!

@g0zilla could you please have a look at the changes I've put onto yours?

g0zilla commented:
@Silvanoc , this is okay from my side.

Silvanoc commented:
> @Silvanoc , this is okay from my side.

I will then squash my micro-commits and mark the PR as ready for merging. Thanks for the quick feedback!

ajcraig reviewed (changes requested):
This workflow looks good and demonstrates how Fido on boarding could assist us in on boarding the API client to the Server within the Workload Fleet Manager. 
- There is a note that FIDO has not been selected, but I it's positive to include as a proposal within the documentation. This is still a decision we need to finalize within the Management interface working group.

As for the content, I would recommend removing the sections in the mermaid drawing beyond line # 57. I mention this as I think we should have a sequence diagram within the workload deployment markdown describing this process. I have a draft of this sequence in the following issue [comment](https://github.com/margo/specification/issues/30#issuecomment-2631897939)

Additionally, lines 55-71 mention the sequence for Git storage and retrieval. This strategy to utilize or mandate git storage was not approved by the TWG. I think keeping line 55 shows a crucial step in onboarding the client, however, we should make a note that it is still under discussion how the client will receive url and any access related tokens/secrets to the location where the desired states will be stored.

phil-abb commented:
@Silvanoc what do we want to do with this PR? It's been here for a while and now it has conflicts.

cc @ajcraig

g0zilla committed:
Modified 'workload-orchestration-edge-onboarding.md' to include a diagram example

Signed-off-by: Andreas Ziller <andreas.ziller@siemens.com>
Signed-off-by: Silvano Cirujano Cuesta <silvano.cirujano-cuesta@siemens.com>

Silvanoc committed:
fix: rename wos to wfm

Workload Orchestration Service (WOS) has been renamed to Workload Fleet
Management (WFM) after this PR was originally created. This commit fixes
it.

Signed-off-by: Silvano Cirujano Cuesta <silvano.cirujano-cuesta@siemens.com>

Silvanoc commented:
> @Silvanoc what do we want to do with this PR? It's been here for a while and now it has conflicts.
> 
> cc @ajcraig

@phil-abb now the conflicts are resolved and even the renaming WOS -> WFM has been done.

Unfortunately doing so I might have "moved" the lines in the diagram and therefore I'm not sure if the lines mentioned by @ajcraig in [this comment](https://github.com/margo/specification/pull/73#pullrequestreview-2613257052) still apply or not.

I oversaw the mention comment from Armand, therefore it was still laying around.

Silvanoc commented:
After a second look at @ajcraig [comment](https://github.com/margo/specification/pull/73#pullrequestreview-2613257052), I think that it does not apply to a conversion of the Draw.IO diagram into Mermaid (what is the scope of this PR). But it is rather a change in the diagram.

I would propose to first merge this diagram conversion as it is, without changes (apart from the WOS->WFM renaming) and then fix the diagram in a separate PR. Fixing the diagram at that point is much easier to review, since it's code which changes can be easily seen on the GitHub WebUI.

phil-abb commented:
@Silvanoc This PR has been approved. Is this something you wanted to merge in?

cc @ajcraig

phil-abb commented:
@Silvanoc This PR has been sitting here for a long time. Can you please either complete it or cancel it if it's no longer relevant?

phil-abb commented:
@Silvanoc - Nothing has been done with this PR in a long time, so I'm going to close it. 

@ajcraig
